### ISSUE TITLE
Module not found

### ISSUE THREAD

**Issue by BayBreezy** [finkinfridom/payload-cloudinary-plugin#5]
Hey,

**Thanks for writing this plugin ❤️.**

When I try to use it in my typescript payload app, i get the error `Cannot find module 'payload-cloudinary-plugin' or its corresponding type declarations.` when I try to import it.

If i import it via `import { mediaManagement } from "payload-cloudinary-plugin/src/index";`, then I get an error when i try to run the app ==> `SyntaxError: Cannot use import statement outside a module`

Am I doing something wrong here?

**Comment by finkinfridom** [comment]
Hey, thanks for using it 😄 
I'm just checking the issue you described and I "simply" forgot the 'build' part.
Let you know when fixed (very soon)

**Comment by BayBreezy** [comment]
Yayy!!! 

Thanks again.. Been waiting on this module :)
I will be waiting on the update.

**Comment by finkinfridom** [comment]
@BayBreezy package 0.0.2 just released on the registry (and it's working now).
Thanks for the patience

**Comment by BayBreezy** [comment]
Hey @finkinfridom 

So I installed the package, did my config and no luck.

<details>
  <summary>Click To See Logs</summary>

  ```bash
WARNING in ./node_modules/cloudinary/lib-es5/utils/index.js 1653:13-32
Critical dependency: the request of a dependency is an expression

WARNING in ./node_modules/cloudinary/lib/utils/index.js 1519:13-32
Critical dependency: the request of a dependency is an expression

ERROR in ./node_modules/cloudinary/lib-es5/api_client/execute_request.js 6:18-40
Module not found: Error: Can't resolve 'querystring' in 'C:\Users\behon\Desktop\payload-cld\node_modules\cloudinary\lib-es5\api_client'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
        - add a fallback 'resolve.fallback: { "querystring": require.resolve("querystring-es3") }'
        - install 'querystring-es3'
If you don't want to include a polyfill, you can use an empty module like this:
        resolve.fallback: { "querystring": false }

ERROR in ./node_modules/cloudinary/lib-es5/api_client/execute_request.js 8:10-24
Module not found: Error: Can't resolve 'url' in 'C:\Users\behon\Desktop\payload-cld\node_modules\cloudinary\lib-es5\api_client'    

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
        - add a fallback 'resolve.fallback: { "url": require.resolve("url/") }'
        - install 'url'
If you don't want to include a polyfill, you can use an empty module like this:
        resolve.fallback: { "url": false }

ERROR in ./node_modules/cloudinary/lib-es5/config.js 13:10-24
Module not found: Error: Can't resolve 'url' in 'C:\Users\behon\Desktop\payload-cld\node_modules\cloudinary\lib-es5'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
        - add a fallback 'resolve.fallback: { "url": require.resolve("url/") }'
        - install 'url'
If you don't want to include a polyfill, you can use an empty module like this:
        resolve.fallback: { "url": false }

ERROR in ./node_modules/cloudinary/lib-es5/upload_stream.js 11:16-43
Module not found: Error: Can't resolve 'stream' in 'C:\Users\behon\Desktop\payload-cld\node_modules\cloudinary\lib-es5'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
        - add a fallback 'resolve.fallback: { "stream": require.resolve("stream-browserify") }'
        - install 'stream-browserify'
If you don't want to include a polyfill, you can use an empty module like this:
        resolve.fallback: { "stream": false }

ERROR in ./node_modules/cloudinary/lib-es5/uploader.js 15:9-22
Module not found: Error: Can't resolve 'fs' in 'C:\Users\behon\Desktop\payload-cld\node_modules\cloudinary\lib-es5'

ERROR in ./node_modules/cloudinary/lib-es5/uploader.js 22:15-41
Module not found: Error: Can't resolve 'stream' in 'C:\Users\behon\Desktop\payload-cld\node_modules\cloudinary\lib-es5'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
        - add a fallback 'resolve.fallback: { "stream": require.resolve("stream-browserify") }'
        - install 'stream-browserify'
If you don't want to include a polyfill, you can use an empty module like this:
        resolve.fallback: { "stream": false }

ERROR in ./node_modules/cloudinary/lib-es5/uploader.js 23:13-27
Module not found: Error: Can't resolve 'url' in 'C:\Users\behon\Desktop\payload-cld\node_modules\cloudinary\lib-es5'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
        - add a fallback 'resolve.fallback: { "url": require.resolve("url/") }'
        - install 'url'
If you don't want to include a polyfill, you can use an empty module like this:
        resolve.fallback: { "url": false }

ERROR in ./node_modules/cloudinary/lib-es5/utils/encoding/sdkAnalytics/getSDKVersions.js 3:9-22
Module not found: Error: Can't resolve 'fs' in 'C:\Users\behon\Desktop\payload-cld\node_modules\cloudinary\lib-es5\utils\encoding\sdkAnalytics'

ERROR in ./node_modules/cloudinary/lib-es5/utils/index.js 16:18-40
Module not found: Error: Can't resolve 'querystring' in 'C:\Users\behon\Desktop\payload-cld\node_modules\cloudinary\lib-es5\utils' 

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
        - add a fallback 'resolve.fallback: { "querystring": require.resolve("querystring-es3") }'
        - install 'querystring-es3'
If you don't want to include a polyfill, you can use an empty module like this:
        resolve.fallback: { "querystring": false }

ERROR in ./node_modules/cloudinary/lib-es5/utils/index.js 17:15-35
Module not found: Error: Can't resolve 'url' in 'C:\Users\behon\Desktop\payload-cld\node_modules\cloudinary\lib-es5\utils'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
        - add a fallback 'resolve.fallback: { "url": require.resolve("url/") }'
        - install 'url'
If you don't want to include a polyfill, you can use an empty module like this:
        resolve.fallback: { "url": false }

ERROR in ./node_modules/cloudinary/lib/api_client/execute_request.js 4:20-42
Module not found: Error: Can't resolve 'querystring' in 'C:\Users\behon\Desktop\payload-cld\node_modules\cloudinary\lib\api_client'
BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
        - add a fallback 'resolve.fallback: { "querystring": require.resolve("querystring-es3") }'
        - install 'querystring-es3'
If you don't want to include a polyfill, you can use an empty module like this:
        resolve.fallback: { "querystring": false }

ERROR in ./node_modules/cloudinary/lib/api_client/execute_request.js 6:12-26
Module not found: Error: Can't resolve 'url' in 'C:\Users\behon\Desktop\payload-cld\node_modules\cloudinary\lib\api_client'        

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
        - add a fallback 'resolve.fallback: { "url": require.resolve("url/") }'
        - install 'url'
If you don't want to include a polyfill, you can use an empty module like this:
        resolve.fallback: { "url": false }

ERROR in ./node_modules/cloudinary/lib/config.js 9:12-26
Module not found: Error: Can't resolve 'url' in 'C:\Users\behon\Desktop\payload-cld\node_modules\cloudinary\lib'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
        - add a fallback 'resolve.fallback: { "url": require.resolve("url/") }'
        - install 'url'
If you don't want to include a polyfill, you can use an empty module like this:
        resolve.fallback: { "url": false }

ERROR in ./node_modules/cloudinary/lib/upload_stream.js 2:18-45
Module not found: Error: Can't resolve 'stream' in 'C:\Users\behon\Desktop\payload-cld\node_modules\cloudinary\lib'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
        - add a fallback 'resolve.fallback: { "stream": require.resolve("stream-browserify") }'
        - install 'stream-browserify'
If you don't want to include a polyfill, you can use an empty module like this:
        resolve.fallback: { "stream": false }

ERROR in ./node_modules/cloudinary/lib/uploader.js 1:11-24
Module not found: Error: Can't resolve 'fs' in 'C:\Users\behon\Desktop\payload-cld\node_modules\cloudinary\lib'

ERROR in ./node_modules/cloudinary/lib/uploader.js 4:17-43
Module not found: Error: Can't resolve 'stream' in 'C:\Users\behon\Desktop\payload-cld\node_modules\cloudinary\lib'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
        - add a fallback 'resolve.fallback: { "stream": require.resolve("stream-browserify") }'
        - install 'stream-browserify'
If you don't want to include a polyfill, you can use an empty module like this:
        resolve.fallback: { "stream": false }

ERROR in ./node_modules/cloudinary/lib/uploader.js 5:15-29
Module not found: Error: Can't resolve 'url' in 'C:\Users\behon\Desktop\payload-cld\node_modules\cloudinary\lib'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
        - add a fallback 'resolve.fallback: { "url": require.resolve("url/") }'
        - install 'url'
If you don't want to include a polyfill, you can use an empty module like this:
        resolve.fallback: { "url": false }

ERROR in ./node_modules/cloudinary/lib/utils/encoding/sdkAnalytics/getSDKVersions.js 1:9-22
Module not found: Error: Can't resolve 'fs' in 'C:\Users\behon\Desktop\payload-cld\node_modules\cloudinary\lib\utils\encoding\sdkAnalytics'

ERROR in ./node_modules/cloudinary/lib/utils/index.js 8:20-42
Module not found: Error: Can't resolve 'querystring' in 'C:\Users\behon\Desktop\payload-cld\node_modules\cloudinary\lib\utils'     

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
        - add a fallback 'resolve.fallback: { "querystring": require.resolve("querystring-es3") }'
        - install 'querystring-es3'
If you don't want to include a polyfill, you can use an empty module like this:
        resolve.fallback: { "querystring": false }

ERROR in ./node_modules/cloudinary/lib/utils/index.js 9:17-37
Module not found: Error: Can't resolve 'url' in 'C:\Users\behon\Desktop\payload-cld\node_modules\cloudinary\lib\utils'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
        - add a fallback 'resolve.fallback: { "url": require.resolve("url/") }'
        - install 'url'
If you don't want to include a polyfill, you can use an empty module like this:
        resolve.fallback: { "url": false }

ERROR in ./node_modules/payload-cloudinary-plugin/dist/services/cloudinaryService.js 55:27-40
Module not found: Error: Can't resolve 'fs' in 'C:\Users\behon\Desktop\payload-cld\node_modules\payload-cloudinary-plugin\dist\services'

webpack compiled with 21 errors and 2 warnings
```
</details>

I even installed `cloudinary` separately and nothing changed.

My `.env` file looks like this

<details>
  <summary>Click To See Env File</summary>

```env
MONGODB_URI=mongodb://localhost/plcld
PAYLOAD_SECRET=fa35908c16d7cf7b7f602ea8
api_key = KEY_HERE
api_secret = SECRET_HERE
cloud_name = NAME_HERE
CLOUDINARY_API_KEY = KEY_HERE
CLOUDINARY_API_SECRET = SECRET_HERE
CLOUDINARY_CLOUD_NAME = NAME_HERE
```

</details>

My `server.ts` file looks like this

<details>
  <summary>Click To See Server File</summary>

```ts
import express from "express";
import payload from "payload";
import { mediaManagement } from "payload-cloudinary-plugin";

require("dotenv").config();
const app = express();

// Redirect root to Admin panel
app.get("/", (_, res) => {
	res.redirect("/admin");
});

app.use(
	mediaManagement(
		{
			api_key: process.env.api_key,
			api_secret: process.env.api_secret,
			cloud_name: process.env.cloud_name,
		},
		{ folder: "payloadstuff/test" }
	)
);

// Initialize Payload
payload.init({
	secret: process.env.PAYLOAD_SECRET,
	mongoURL: process.env.MONGODB_URI,
	express: app,
	onInit: () => {
		payload.logger.info(`Payload Admin URL: ${payload.getAdminURL()}`);
	},
});

// Add your own express routes here

app.listen(3000);

```
</details>

My `payload.config.ts` file looks like this

<details>
  <summary>Click To See Config File</summary>

```ts
import { buildConfig } from "payload/config";
import { cloudinaryPlugin } from "payload-cloudinary-plugin";
import path from "path";
// import Examples from './collections/Examples';
import Users from "./collections/Users";
import Media from "./collections/Media";

export default buildConfig({
	serverURL: "http://localhost:3000",
	admin: {
		user: Users.slug,
	},
	collections: [
		Users,
		Media,
		// Add Collections here
		// Examples,
	],
	typescript: {
		outputFile: path.resolve(__dirname, "payload-types.ts"),
	},
	graphQL: {
		schemaOutputFile: path.resolve(__dirname, "generated-schema.graphql"),
	},
	plugins: [cloudinaryPlugin()],
});


```
</details>

**Comment by finkinfridom** [comment]
That's very strange but I'm trying to reproduce it locally.

**Comment by finkinfridom** [comment]
I found out the issue. 
If you change the cloudinaryPlugin import from

`import { cloudinaryPlugin } from "payload-cloudinary-plugin";`

to

`import cloudinaryPlugin from "payload-cloudinary-plugin/dist/plugins/cloudinaryPlugin";`

it will start working as expected.

In the meantime, I'll fix how the cloudinaryPlugin is exported.

**Comment by finkinfridom** [comment]
package 0.0.3 just released.
The import statement for the `cloudinaryPlugin` could be set as

`import cloudinaryPlugin from "payload-cloudinary-plugin/dist/plugins";`

**Comment by BayBreezy** [comment]
Hey @finkinfridom , thanks for the update.

So i was able to start the app. I ran into another error tho. If I set `disableLocalStorage: true,` in my Media collection, the image is stored locally but not uploaded to cloudinary. I get this error message in the Toast popup and in the console

```bash
APIError: Cloudinary: {"errno":-4058,"code":"ENOENT","syscall":"stat","path":"C:\\Users\\behon\\Desktop\\plcld\\src\\media\\pexels-wendy-wei-1656684.jpg"}
```

If I set disableLocalStorage to false in my Media config, it works just fine.

**Comment by finkinfridom** [comment]
hey @BayBreezy, the issue has been spotted. 
I introduced a custom logic to allow users to upload the same image several times without "replacing" the first one. 
Let's imagine you and your colleague, working on the Media collection. It may happen, when data is coming from external sources, that different images have equal names (e.g. DSC24641.png) and would lead to replacing "incorrect" files (based on upload time).
I am adding some tests there to understand what I'm missing.
let you know when fixed

**Comment by finkinfridom** [comment]
Hey @BayBreezy, the issue has been fixed. Incredibly easy to fix and now version 0.0.5 is available

**Comment by BayBreezy** [comment]
Hey @finkinfridom ,

Problem still there. I think the issue is in the [cloudinaryService.ts](https://github.com/finkinfridom/payload-cloudinary-plugin/blob/main/src/services/cloudinaryService.ts#L52) 

I think it's cause you are removing the file before the `beforeChange` hook is called? (I may be wrong)

**Comment by BayBreezy** [comment]
Hey @finkinfridom ,

Thanks for the update. So now.. the issue is that images are being uploaded but no other file is accepted. Was this intentional?
Even if I add the allowed formats in an array like so `allowed_formats: ["png", "jpg", "jpeg", "pdf", "csv", "xlsx"]` I still get the error : `APIError: Cloudinary: {"message":"Invalid image file","name":"Error","http_code":400}`

**Comment by finkinfridom** [comment]
It seems we're not alone https://stackoverflow.com/a/32887402/13389907 and there's the possibility to let Cloudinary guess which format is uploaded.
I think this needs a bit more development (as intelligent guessing always scares me :))

Do you mind opening a new issue with the above? so it will be easier to track it down.

Thanks
